src/cargo/ops/cargo_compile.rs

```diff
@@ -106,26 +106,23 @@ pub enum MessageFormat {
 
 #[derive(Clone, Copy, PartialEq, Eq, Debug)]
 pub enum Packages<'a> {
+    Default,
     All,
     OptOut(&'a [String]),
     Packages(&'a [String]),
 }
 
 impl<'a> Packages<'a> {
-    pub fn from_flags(virtual_ws: bool, all: bool, exclude: &'a [String], package: &'a [String])
+    pub fn from_flags(all: bool, exclude: &'a [String], package: &'a [String])
         -> CargoResult<Self>
     {
-        let all = all || (virtual_ws && package.is_empty());
```

Comment:
Bikeshedding: this could be `match (all, exclude.is_empty(), package.is_empty())` and `(false, true, true)...`

Reply:
I did that at first and found it confusing when it came to write all the patterns. Both because too many booleans, and because `false` felt like "negative" and so "not present".